osh/hub/osh_xmlrpc/worker.py

```diff
@@ -76,6 +79,26 @@ def finish_task(request, task_id):
                 task.fail_task()
 
 
+@validate_worker
+def open_task(request, task_id):
+    response = kobo_open_task(request, task_id)
+
+    if settings.ENABLE_SINGLE_USE_WORKERS:
+        task = Task.objects.get(id=task_id)
+
+        # Do not set `max_load` to 0 in that case, as the base scan would get assigned but never
+        # start as `max_load` has been set to 0.
+        if task.method == "VersionDiffBuild":
+            pass
+        else:
+            # Set the worker load to 0. This should avoid getting any new tasks assigned the worker.
+            # Each single use worker should be destroyed once a task completes.
+            task.worker.max_load = 0
```

Comment:
Shall we be setting this field when the task is closed, interrupted or failed? Could there be any race condition there i.e. a worker may pick up a task before `max_load` is set to `0`.